Docker/yaypatch.patch

```diff
@@ -0,0 +1,11 @@
+--- Expansion.php  2019-04-16 18:58:13.254005966 +0000
++++ Expansion.php       2019-04-16 18:59:03.202020632 +0000
+@@ -380,6 +380,9 @@
+ 
+                     $delimiters = $result->{'delimiters'};
+ 
++                    // HACK ALERT: THIS SHOULDN'T HAPPEN AND SHOULD BE FIXED UPSTREAM
++                    if (!is_array($context)) $context = [$context];
```

Comment:
/o\ if you have a test, please let me know so we can put it as a `.phpt` file 👍

Reply:
@marcioAlmada I don't have one, other than trying to compile this repo with `scripts/rebuild.php`. JIT.pre is one that always fails to compile for me, but I don't know what causes it to fail enough to turn it into a useful test case.

Reply:
Unfortunately I haven't been able to reproduce. Sometimes it happens, sometimes it doesn't. Really quite random...

Reply:
And by random, I mean changes to macro.yay sometimes cause it, sometimes don't. I haven't found a clear pattern in order to be able to reproduce...

Reply:
Oh, so it depends on what is being matched. I'll try to poke the call graph.

Reply:
@marcioAlmada if you want to poke at it, check out the master branch, run composer install (manually, without the patch), then run scripts/rebuild.php. In all likelyhood strlen.pre will work and then JIT.pre will fail (at least that's been the case for most environments I've tried it in)

Reply:
So, the macro has a bug. An `Ast` node holding a single token was being unpacked, resulting in a scope which was not an array. At  `$(binary_xor ? ... {` the scope became `Token("^")` instead of `["binary_xor" => Token("^")]`. The correct macro would be:

```diff
$(stmts ... {
    $(assignop ? ... {
        $(binary ? ... {
            $(binary_op ... {
-                $(binary_xor ? ... {
+                $(binary_xor ? {
                    $(result) = $this->context->builder->bitwiseXor($(binary_left), $__right);
                })
            })
        })
    })
})
```

But... of course, untreated PHP error is not fine. I'm adding a runtime error when attempting to unpack an `Ast` node holding a non unpackable value :+1:

Reply:
Done. From now, when a situation like this is encountered, the following runtime error will be thrown:

```
Error unpacking non unpackable Ast node on `$(binary_xor ...? {` at line 29 context: [
    "^"
]

Hint: use a non ellipsis expansion as in `$(binary_xor ? {`
```

Reply:
BTW this was weird to debug due to dependencies being hidden...

Reply:
yeah, but at least https://github.com/preprocess/pre-plugin/pull/18 aims to fix that...

Reply:
I'm now seeing a bunch of warnings: Warning: count(): Parameter must be an array or an object that implements Countable in /compiler/vendor/yay/yay/src/Ast.php on line 136

But it appears to be successful...

Reply:
Are you sure you fixed the macro? That sounds like the same warning as before when there was a problem

Reply:
Some fixes were done at yay/master branch too, so I guess you also have to composer update the deps.

Reply:
These are the changes that had to be done in your macros: https://gist.github.com/marcioAlmada/6f07c9b0ef7297c9ee82f91bc2a53047